source/common/router/router.cc

```diff
@@ -600,11 +600,19 @@ std::unique_ptr<GenericConnPool> Filter::createConnPool() {
   const bool should_tcp_proxy =
       route_entry_->connectConfig().has_value() &&
       downstream_headers_->getMethodValue() == Http::Headers::get().MethodValues.Connect;
-
+  Http::Protocol protocol = cluster_->upstreamHttpProtocol(callbacks_->streamInfo().protocol());
   if (should_tcp_proxy) {
-    return std::make_unique<TcpConnPool>();
+    auto pool = std::make_unique<TcpConnPool>(config_.cm_, *route_entry_, protocol, this);
+    if (pool->valid()) {
```

Comment:
This will eventually become a factory that can return null, right? If so this makes sense to me as a stop gap. Another option would be to just create a static factory for each type that can return null in this PR. Up to you.

Reply:
Exactly - the factory will return null - setting that PR up now :-)

Reply:
Cool sounds good.